### PR TITLE
modulegroup fix : quick way to add modules

### PR DESCRIPTION
**Pull request by AlicVB**
As pointed by some of you, new modulegroups is way more flexible but as a big issue, especially for "old" users workflows : it's now way more difficult to quickly show modules for temporary use.

This PR allow to right-click on the group button to show a menu with all modules to add.
Internally, when adding a module this way, a temporary preset named "last modified layout" is created on the fly.

I know we are actual in feature freeze, but imho, this clearly solve an issue as current state breaks some legacy workflows. Codewise, the change are imo quite reduced, safe and not intrusive at all...

### CODE REVIEW

**Review by aurelienpierre — APPROVED** [review]
That sounds like a better, however incomplete, way to revert the user-experience regression induced by the layout change. At least now we have a quick way to show modules in UI. Last step will be a quick way to remove them from UI.

Test shows no runtime problem. I will let Pascal check for the Gtk validity and memory leaks ;-)

**Comment by AlicVB** [comment]
@aurelienpierre : add+remove is already done here. But I prefer to postpone that for 3.4.1... I'll do a PR once this one is merged to keep git happy ;)

**Comment by TurboGit** [comment]
I think there is something wrong. Select the default preset "workflow: scene-referred". Right click on basic button the list seems ok with the recommended modules first and the other after. Likewise for all buttons except the last one where this is no section.

**Comment by TurboGit** [comment]
Or maybe just because there is no recommended modules for this group ?

**Comment by AlicVB** [comment]
hmm... yes, if no section is shown, that means that there's no recommended modules to add. Either they are all here already, either no recommended modules are bound to the group. That's already like that in the editor...

What is strange, is that here, with preset "workflow: scene-referred" that doesn't happen, I have recommended modules for all groups... With "modules: default", I have exactly what you mentioned... are you sure you can reproduce with built-in "workflow: scene refereed" preset ?

**Comment by TurboGit** [comment]
Yes, I can reproduce all the time with "workflow: scene-referred" & ""workflow: display-referred". It works as expected for all other presets. For "modules: all" I can reproduce but here it is expected and the same for all groups.

BTH, this new menu should probably be deactivated for the "modules: deprecated" where it makes no sense.

**Comment by TurboGit** [comment]
I confirm, just tested with a memory library and clean darktablerc file and same issue for both scene & display referred presets.

**Comment by TurboGit** [comment]
As we are at it, can we put an 's' to others in line 1433:

`    lb = gtk_label_new(_("others"));`

**Comment by TurboGit** [comment]
> As we are at it, can we put an 's' to others in line 1433:
> 
> ` lb = gtk_label_new(_("others"));`

Or better use a translation taking into account the plural. I'll look at this in a separate PR.

**Comment by AlicVB** [comment]
> Or better use a translation taking into account the plural

is it worse the pain ? singular would mean that *all-1* existing modules have been added to one unique group. Not sure it will happen...

**Comment by TurboGit** [comment]
> is it worse the pain ? singular would mean that _all-1_ existing modules have been added to one unique group. Not sure it will happen...

Not sure but "recommended" in English is a common word and we should in French translate it as "recommandés" which may be wrong in other part of the code. So at the end I have also used "recommended module" (with plural form) as it was not more costly than that.

**Comment by TurboGit** [comment]
@AlicVB : see #6972

**Comment by AlicVB** [comment]
OK, your other bug was a nice one : that was a translation problem !
well, in fact, in the code we used C_("modulegroup", "effect") in some places and _("effect") in some other... everything changed to contextual translation and force pushed !

File: src/libs/modulegroups.c
Comment (TurboGit) [inline src/libs/modulegroups.c:1409]: Ah that was it ! Now the question is how could this have been working for you :( Some code are mystery to me :)

Comment (AlicVB) [reply src/libs/modulegroups.c:1409]: because I mostly use dt in English...

Comment (TurboGit) [reply src/libs/modulegroups.c:1409]: All clear then !

**Comment by TurboGit** [comment]
Seems good now, I'll wait for the GitLab-CI full report and will merge if all goes well.

We will need a bit of CSS from @Nilvus for this new menu which is missing a border.

**Comment by AlicVB** [comment]
this menu is not really new as it's already used inside the layout manager, but yes some css is certainly needed !

**Comment by TurboGit** [comment]
Thanks !

**Comment by Nilvus** [comment]
> Seems good now, I'll wait for the GitLab-CI full report and will merge if all goes well.
> 
> We will need a bit of CSS from @Nilvus for this new menu which is missing a border.

All other popup menus in darktable never had a border. This menu needs at least a padding. I'm not against adding a border but I think is not necessary with a little padding. But if we add a border there, that would mean for consistency a lot of others popup menu (preset, styles in darkroom, gamut, display, etc) to have same borders.

**Comment by TurboGit** [comment]
@Nilvus : Well there is a border in fact for grey & darker theme, but for dark there is no and so the content of the menu is not delimited. See:

dark:

image

darker:

image

**Comment by TurboGit** [comment]
And BTW, the border is maybe a bit too visible a thinner may be better. And yes, padding (as you've already proposed is needed).

**Comment by Nilvus** [comment]
@TurboGit: strange, but other menus (check presets ones, or styles on bottom bar in darkroom) doesn't have borders. I will check that ASAP.

**Pull request merged**
